buildTar fails with latest Google distroless Java 17 images

**Environment**:

- *Jib version:* 3.3.1
- *Build tool:* maven, 3.6.6
- *OS:* Linux, Ubuntu

```
Apache Maven 3.6.3
Maven home: /usr/share/maven
Java version: 17.0.6, vendor: Private Build, runtime: /usr/lib/jvm/java-17-openjdk-amd64
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "5.15.0-67-generic", arch: "amd64", family: "unix"
```


**Description of the issue**:

We updated our base images (hash is the difference) from

`gcr.io/distroless/java17-debian11:nonroot@sha256:3481476327255f2e553415945d9c774dc5936273bf94cfc01509f044da7b3ed5`

to

`gcr.io/distroless/java17-debian11:nonroot@sha256:5154596f58d81687d66c8302face284198caf65033cde6bf6667c488b21745e6`

The maven build fails with

`[ERROR] Failed to execute goal com.google.cloud.tools:jib-maven-plugin:3.3.1:buildTar (default-cli) on project spring-boot-k8s-example: class com.google.cloud.tools.jib.image.json.OciIndexTemplate cannot be cast to class com.google.cloud.tools.jib.image.json.V22ManifestListTemplate (com.google.cloud.tools.jib.image.json.OciIndexTemplate and com.google.cloud.tools.jib.image.json.V22ManifestListTemplate are in unnamed module of loader org.codehaus.plexus.classworlds.realm.ClassRealm @2262d6d5) -> [Help 1]
`

**Expected behavior**:

Build passes as before and our application images are build successfully.

**Steps to reproduce**:
<!-- Please provide a minimal and precise series of steps -->

  1. Download the spring boot example from this repo
  2. Add the below pom.xml snippet to the jib plugin definition
  3. Run `./mvnw -DskipTestspackage jib:buildTar`

**`jib-maven-plugin` Configuration**: <!-- Delete this section if not used -->
```xml
<plugin>
  <groupId>com.google.cloud.tools</groupId>
  <artifactId>jib-maven-plugin</artifactId>
  <version>3.3.1</version>
  <configuration>
    <from>
      <image>gcr.io/distroless/java17-debian11:nonroot@sha256:5154596f58d81687d66c8302face284198caf65033cde6bf6667c488b21745e6</image>
    </from>
  </configuration>
</plugin>
```
**Log output**: <!-- If applicable, provide relevant log output -->

```
$ ./mvnw -e package jib:buildTar
[INFO] Error stacktraces are turned on.
[INFO] Scanning for projects...
[INFO] 
[INFO] ------------------< example:spring-boot-k8s-example >-------------------
[INFO] Building spring-boot-k8s-example 0.1.0
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-resources-plugin:3.1.0:resources (default-resources) @ spring-boot-k8s-example ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /home/fziegler/3rd-party/jib/examples/spring-boot/src/main/resources
[INFO] skip non existing resourceDirectory /home/fziegler/3rd-party/jib/examples/spring-boot/src/main/resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.1:compile (default-compile) @ spring-boot-k8s-example ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- maven-resources-plugin:3.1.0:testResources (default-testResources) @ spring-boot-k8s-example ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /home/fziegler/3rd-party/jib/examples/spring-boot/src/test/resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.1:testCompile (default-testCompile) @ spring-boot-k8s-example ---
[INFO] No sources to compile
[INFO] 
[INFO] --- maven-surefire-plugin:2.22.2:test (default-test) @ spring-boot-k8s-example ---
[INFO] No tests to run.
[INFO] 
[INFO] --- maven-jar-plugin:3.1.2:jar (default-jar) @ spring-boot-k8s-example ---
[INFO] 
[INFO] --- jib-maven-plugin:3.3.1:buildTar (default-cli) @ spring-boot-k8s-example ---
[INFO] Tagging image with generated image reference spring-boot-k8s-example:0.1.0. If you'd like to specify a different tag, you can set the <to><image> parameter in your pom.xml, or use the -Dimage=<MY IMAGE> commandline flag.
[WARNING] 'mainClass' configured in 'maven-jar-plugin' is not a valid Java class: ${start-class}
[INFO] 
[INFO] Containerizing application to file at '/home/fziegler/3rd-party/jib/examples/spring-boot/target/jib-image.tar'...
[INFO] Executing tasks:
[INFO] [============                  ] 40.0% complete
[INFO] > building image to tar file
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 2.439 s
[INFO] Finished at: 2023-03-15T16:08:53+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.google.cloud.tools:jib-maven-plugin:3.3.1:buildTar (default-cli) on project spring-boot-k8s-example: class com.google.cloud.tools.jib.image.json.OciIndexTemplate cannot be cast to class com.google.cloud.tools.jib.image.json.V22ManifestListTemplate (com.google.cloud.tools.jib.image.json.OciIndexTemplate and com.google.cloud.tools.jib.image.json.V22ManifestListTemplate are in unnamed module of loader org.codehaus.plexus.classworlds.realm.ClassRealm @670f2466) -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.google.cloud.tools:jib-maven-plugin:3.3.1:buildTar (default-cli) on project spring-boot-k8s-example: class com.google.cloud.tools.jib.image.json.OciIndexTemplate cannot be cast to class com.google.cloud.tools.jib.image.json.V22ManifestListTemplate (com.google.cloud.tools.jib.image.json.OciIndexTemplate and com.google.cloud.tools.jib.image.json.V22ManifestListTemplate are in unnamed module of loader org.codehaus.plexus.classworlds.realm.ClassRealm @670f2466)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:213)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:954)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.apache.maven.wrapper.BootstrapMainStarter.start (BootstrapMainStarter.java:39)
    at org.apache.maven.wrapper.WrapperExecutor.execute (WrapperExecutor.java:122)
    at org.apache.maven.wrapper.MavenWrapperMain.main (MavenWrapperMain.java:55)
Caused by: org.apache.maven.plugin.MojoExecutionException: class com.google.cloud.tools.jib.image.json.OciIndexTemplate cannot be cast to class com.google.cloud.tools.jib.image.json.V22ManifestListTemplate (com.google.cloud.tools.jib.image.json.OciIndexTemplate and com.google.cloud.tools.jib.image.json.V22ManifestListTemplate are in unnamed module of loader org.codehaus.plexus.classworlds.realm.ClassRealm @670f2466)
    at com.google.cloud.tools.jib.maven.BuildTarMojo.execute (BuildTarMojo.java:159)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:954)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.apache.maven.wrapper.BootstrapMainStarter.start (BootstrapMainStarter.java:39)
    at org.apache.maven.wrapper.WrapperExecutor.execute (WrapperExecutor.java:122)
    at org.apache.maven.wrapper.MavenWrapperMain.main (MavenWrapperMain.java:55)
Caused by: java.lang.ClassCastException: class com.google.cloud.tools.jib.image.json.OciIndexTemplate cannot be cast to class com.google.cloud.tools.jib.image.json.V22ManifestListTemplate (com.google.cloud.tools.jib.image.json.OciIndexTemplate and com.google.cloud.tools.jib.image.json.V22ManifestListTemplate are in unnamed module of loader org.codehaus.plexus.classworlds.realm.ClassRealm @670f2466)
    at com.google.cloud.tools.jib.builder.steps.PullBaseImageStep.getCachedBaseImages (PullBaseImageStep.java:470)
    at com.google.cloud.tools.jib.builder.steps.PullBaseImageStep.call (PullBaseImageStep.java:134)
    at com.google.cloud.tools.jib.builder.steps.PullBaseImageStep.call (PullBaseImageStep.java:71)
    at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly (TrustedListenableFutureTask.java:131)
    at com.google.common.util.concurrent.InterruptibleTask.run (InterruptibleTask.java:74)
    at com.google.common.util.concurrent.TrustedListenableFutureTask.run (TrustedListenableFutureTask.java:82)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1136)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:635)
    at java.lang.Thread.run (Thread.java:833)
[ERROR] 
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

**Additional Information**: <!-- Any additional information that may be helpful -->

The behavior does not occur with the previously mentioned image  (`gcr.io/distroless/java17-debian11:nonroot@sha256:3481476327255f2e553415945d9c774dc5936273bf94cfc01509f044da7b3ed5`) .

There was a change to the image build that moved from docker manifest format to OCI format

```
docker buildx imagetools inspect gcr.io/distroless/java17-debian11:nonroot@sha256:3481476327255f2e553415945d9c774dc5936273bf94cfc01509f044da7b3ed5
Name:      gcr.io/distroless/java17-debian11:nonroot@sha256:3481476327255f2e553415945d9c774dc5936273bf94cfc01509f044da7b3ed5
MediaType: application/vnd.docker.distribution.manifest.list.v2+json
Digest:    sha256:3481476327255f2e553415945d9c774dc5936273bf94cfc01509f044da7b3ed5
           
Manifests: 
  Name:      gcr.io/distroless/java17-debian11:nonroot@sha256:8d3f355ee92b8da899db8739ad994740cace766fa5c22abce49ba242657157f2
  MediaType: application/vnd.docker.distribution.manifest.v2+json
  Platform:  linux/amd64
             
  Name:      gcr.io/distroless/java17-debian11:nonroot@sha256:01022fc221d78827d9d2f7d021eaad6705e50b1983fe04e45606c75758b72ed0
  MediaType: application/vnd.docker.distribution.manifest.v2+json
  Platform:  linux/arm64
             
  Name:      gcr.io/distroless/java17-debian11:nonroot@sha256:b4de50145c10fed2067f1125ab7bdc0dd5468afa0e27807e01c5e86145d1289c
  MediaType: application/vnd.docker.distribution.manifest.v2+json
  Platform:  linux/ppc64le
             
  Name:      gcr.io/distroless/java17-debian11:nonroot@sha256:b337804c4852f00ff55277c44c889bcf352384082e61bb557bfb61f0d56c8f68
  MediaType: application/vnd.docker.distribution.manifest.v2+json
  Platform:  linux/s390x
```

```
docker buildx imagetools inspect gcr.io/distroless/java17-debian11:nonroot@sha256:5154596f58d81687d66c8302face284198caf65033cde6bf6667c488b21745e6
Name:      gcr.io/distroless/java17-debian11:nonroot@sha256:5154596f58d81687d66c8302face284198caf65033cde6bf6667c488b21745e6
MediaType: application/vnd.oci.image.index.v1+json
Digest:    sha256:5154596f58d81687d66c8302face284198caf65033cde6bf6667c488b21745e6
           
Manifests: 
  Name:      gcr.io/distroless/java17-debian11:nonroot@sha256:4f1f85307f592125bae41a366e62e9c0e7c916e053854004b3acf1cbf7c726e4
  MediaType: application/vnd.oci.image.manifest.v1+json
  Platform:  linux/amd64
             
  Name:      gcr.io/distroless/java17-debian11:nonroot@sha256:b3fb74abd29a0a15b6a96f0ce3b1b10fe8cd659f5b3a34a68eda7940901a7cc2
  MediaType: application/vnd.oci.image.manifest.v1+json
  Platform:  linux/arm64
             
  Name:      gcr.io/distroless/java17-debian11:nonroot@sha256:33fdfda1fb7c109ea3cfa239ad37a1d4545e7fc3ce08a8d9592f5054882640b5
  MediaType: application/vnd.oci.image.manifest.v1+json
  Platform:  linux/s390x
             
  Name:      gcr.io/distroless/java17-debian11:nonroot@sha256:0616be507c506457ba61a5204595860a4709b9c7e665d35173172d4bbe7cf8bc
  MediaType: application/vnd.oci.image.manifest.v1+json
  Platform:  linux/ppc64le
```

See this commit:

https://github.com/GoogleContainerTools/distroless/commit/fb94acc429d87523020a0a801446f178708e2a10

<!-- Thanks for contributing! -->

I get similar behavior but with a different error message:

<details>
```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  02:48 min
[INFO] Finished at: 2023-03-26T06:18:20Z
[INFO] ------------------------------------------------------------------------
Error:  Failed to execute goal com.google.cloud.tools:jib-maven-plugin:3.2.1:build (default-cli) on project pocketbook-java: Tried to pull image manifest for gcr.io/distroless/java17-debian11:latest but failed because: Manifest with tag 'latest' has media type 'application/vnd.oci.image.index.v1+json', but client accepts 'application/vnd.oci.image.manifest.v1+json,application/vnd.docker.distribution.manifest.v2+json,application/vnd.docker.distribution.manifest.v1+json,application/vnd.docker.distribution.manifest.list.v2+json'. 404 Not Found
Error: [ERROR] ***"errors":[***"code":"MANIFEST_UNKNOWN","message":"Manifest with tag 'latest' has media type 'application/vnd.oci.image.index.v1+json', but client accepts 'application/vnd.oci.image.manifest.v1+json,application/vnd.docker.distribution.manifest.v2+json,application/vnd.docker.distribution.manifest.v1+json,application/vnd.docker.distribution.manifest.list.v2+json'."***]***
Error:  -> [Help 1]
Error:  
Error:  To see the full stack trace of the errors, re-run Maven with the -e switch.
Error:  Re-run Maven using the -X switch to enable full debug logging.
Error:  
Error:  For more information about the errors and possible solutions, please read the following articles:
Error:  [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
Error: Process completed with exit code 1.
```
</details>

--- a/jib-core/examples/build.gradle/README.md
+++ b/jib-core/examples/build.gradle/README.md
@@ -10,10 +10,11 @@ For example, the following snippet is a simple example that creates a Gradle tas
 // Imports Jib Core as a library to use in this build script.
 buildscript {
   repositories {
+    mavenLocal()
     mavenCentral()
   }
   dependencies {
-    classpath 'com.google.cloud.tools:jib-core:0.20.0'
+    classpath 'com.google.cloud.tools:jib-core:0.23.0'
   }
 }
 
@@ -41,7 +42,9 @@ task('dojib') {
                     // Tells Jib to get registry credentials from a Docker config.
                     .addCredentialRetriever(
                         CredentialRetrieverFactory
-                            .forImage(ImageReference.parse(targetImage))
+                            .forImage(
+                                    ImageReference.parse(targetImage),
+                                    logEvent -> logger.log(LogLevel.valueOf(logEvent.getLevel().name()), logEvent.getMessage()))
                             .dockerConfig())))
     println 'done'
   }

--- a/jib-core/src/integration-test/java/com/google/cloud/tools/jib/api/JibIntegrationTest.java
+++ b/jib-core/src/integration-test/java/com/google/cloud/tools/jib/api/JibIntegrationTest.java
@@ -49,6 +49,10 @@ import org.junit.rules.TemporaryFolder;
 /** Integration tests for {@link Jib}. */
 public class JibIntegrationTest {
 
+  /** A known oci index sha for gcr.io/distroless/base. */
+  public static final String KNOWN_OCI_INDEX_SHA =
+      "sha256:2c50b819aa3bfaf6ae72e47682f6c5abc0f647cf3f4224a4a9be97dd30433909";
+
   @ClassRule public static final LocalRegistry localRegistry = new LocalRegistry(5000);
 
   @Rule public final TemporaryFolder temporaryFolder = new TemporaryFolder();
@@ -61,6 +65,14 @@ public class JibIntegrationTest {
               EventHandlers.NONE,
               dockerHost + ":5000",
               "jib-scratch",
+              new FailoverHttpClient(true, true, ignored -> {}))
+          .newRegistryClient();
+
+  private final RegistryClient distrolessRegistryClient =
+      RegistryClient.factory(
+              EventHandlers.NONE,
+              dockerHost + ":5000",
+              "jib-distroless",
               new FailoverHttpClient(true, true, ignored -> {}))
           .newRegistryClient();
 
@@ -290,6 +302,33 @@ public class JibIntegrationTest {
     Assert.assertEquals("windows", platform1.getOs());
     Assert.assertEquals("amd32", platform2.getArchitecture());
     Assert.assertEquals("windows", platform2.getOs());
+  }
+
+  @Test
+  public void testDistroless_ociManifest()
+      throws IOException, InterruptedException, ExecutionException, RegistryException,
+          CacheDirectoryCreationException, InvalidImageReferenceException {
+    Jib.from("gcr.io/distroless/base@" + KNOWN_OCI_INDEX_SHA)
+        .setPlatforms(
+            ImmutableSet.of(new Platform("arm64", "linux"), new Platform("amd64", "linux")))
+        .containerize(
+            Containerizer.to(
+                    RegistryImage.named(dockerHost + ":5000/jib-distroless:multi-platform"))
+                .setAllowInsecureRegistries(true));
+
+    V22ManifestListTemplate manifestList =
+        (V22ManifestListTemplate)
+            distrolessRegistryClient.pullManifest("multi-platform").getManifest();
+    Assert.assertEquals(2, manifestList.getManifests().size());
+    ManifestDescriptorTemplate.Platform platform1 =
+        manifestList.getManifests().get(0).getPlatform();
+    ManifestDescriptorTemplate.Platform platform2 =
+        manifestList.getManifests().get(1).getPlatform();
+
+    Assert.assertEquals("arm64", platform1.getArchitecture());
+    Assert.assertEquals("linux", platform1.getOs());
+    Assert.assertEquals("amd64", platform2.getArchitecture());
+    Assert.assertEquals("linux", platform2.getOs());
   }
 
   @Test

--- a/jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/PullBaseImageStep.java
+++ b/jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/PullBaseImageStep.java
@@ -47,7 +47,6 @@ import com.google.cloud.tools.jib.image.json.PlatformNotFoundInBaseImageExceptio
 import com.google.cloud.tools.jib.image.json.UnknownManifestFormatException;
 import com.google.cloud.tools.jib.image.json.UnlistedPlatformInManifestListException;
 import com.google.cloud.tools.jib.image.json.V21ManifestTemplate;
-import com.google.cloud.tools.jib.image.json.V22ManifestListTemplate;
 import com.google.cloud.tools.jib.json.JsonTemplateMapper;
 import com.google.cloud.tools.jib.registry.ManifestAndDigest;
 import com.google.cloud.tools.jib.registry.RegistryClient;
@@ -468,7 +467,7 @@ class PullBaseImageStep implements Callable<ImagesAndRegistryClient> {
     ImmutableList.Builder<Image> images = ImmutableList.builder();
     for (Platform platform : buildContext.getContainerConfiguration().getPlatforms()) {
       String manifestDigest =
-          lookUpPlatformSpecificImageManifest((V22ManifestListTemplate) manifestList, platform);
+          lookUpPlatformSpecificImageManifest((ManifestListTemplate) manifestList, platform);
 
       Optional<ManifestAndConfigTemplate> manifestAndConfigFound =
           manifestsAndConfigs.stream()

--- a/jib-core/src/test/java/com/google/cloud/tools/jib/builder/steps/PullBaseImageStepTest.java
+++ b/jib-core/src/test/java/com/google/cloud/tools/jib/builder/steps/PullBaseImageStepTest.java
@@ -38,9 +38,11 @@ import com.google.cloud.tools.jib.image.Image;
 import com.google.cloud.tools.jib.image.LayerCountMismatchException;
 import com.google.cloud.tools.jib.image.LayerPropertyNotFoundException;
 import com.google.cloud.tools.jib.image.json.BadContainerConfigurationFormatException;
+import com.google.cloud.tools.jib.image.json.BuildableManifestTemplate;
 import com.google.cloud.tools.jib.image.json.ContainerConfigurationTemplate;
 import com.google.cloud.tools.jib.image.json.ImageMetadataTemplate;
 import com.google.cloud.tools.jib.image.json.ManifestAndConfigTemplate;
+import com.google.cloud.tools.jib.image.json.ManifestListTemplate;
 import com.google.cloud.tools.jib.image.json.ManifestTemplate;
 import com.google.cloud.tools.jib.image.json.OciIndexTemplate;
 import com.google.cloud.tools.jib.image.json.PlatformNotFoundInBaseImageException;
@@ -350,7 +352,7 @@ public class PullBaseImageStepTest {
   }
 
   @Test
-  public void testGetCachedBaseImages_v22ManifestCached()
+  public void testGetCachedBaseImages_manifestCached()
       throws InvalidImageReferenceException, IOException, CacheCorruptedException,
           UnlistedPlatformInManifestListException, BadContainerConfigurationFormatException,
           LayerCountMismatchException, PlatformNotFoundInBaseImageException {
@@ -363,7 +365,7 @@ public class PullBaseImageStepTest {
     containerConfigJson.setOs("fat system");
     ManifestAndConfigTemplate manifestAndConfig =
         new ManifestAndConfigTemplate(
-            new V22ManifestTemplate(), containerConfigJson, "sha256:digest");
+            Mockito.mock(BuildableManifestTemplate.class), containerConfigJson, "sha256:digest");
     ImageMetadataTemplate imageMetadata =
         new ImageMetadataTemplate(null, Arrays.asList(manifestAndConfig));
     Mockito.when(cache.retrieveMetadata(imageReference)).thenReturn(Optional.of(imageMetadata));
@@ -377,7 +379,7 @@ public class PullBaseImageStepTest {
   }
 
   @Test
-  public void testGetCachedBaseImages_v22ManifestListCached()
+  public void testGetCachedBaseImages_manifestListCached()
       throws InvalidImageReferenceException, IOException, CacheCorruptedException,
           UnlistedPlatformInManifestListException, BadContainerConfigurationFormatException,
           LayerCountMismatchException, PlatformNotFoundInBaseImageException {
@@ -390,7 +392,7 @@ public class PullBaseImageStepTest {
     containerConfigJson1.setContainerUser("user1");
     containerConfigJson2.setContainerUser("user2");
 
-    V22ManifestListTemplate manifestList = Mockito.mock(V22ManifestListTemplate.class);
+    ManifestListTemplate manifestList = Mockito.mock(ManifestListTemplate.class);
     Mockito.when(manifestList.getDigestsForPlatform("arch1", "os1"))
         .thenReturn(Arrays.asList("sha256:digest1"));
     Mockito.when(manifestList.getDigestsForPlatform("arch2", "os2"))
@@ -401,9 +403,13 @@ public class PullBaseImageStepTest {
             manifestList,
             Arrays.asList(
                 new ManifestAndConfigTemplate(
-                    new V22ManifestTemplate(), containerConfigJson1, "sha256:digest1"),
+                    Mockito.mock(BuildableManifestTemplate.class),
+                    containerConfigJson1,
+                    "sha256:digest1"),
                 new ManifestAndConfigTemplate(
-                    new V22ManifestTemplate(), containerConfigJson2, "sha256:digest2")));
+                    Mockito.mock(BuildableManifestTemplate.class),
+                    containerConfigJson2,
+                    "sha256:digest2")));
     Mockito.when(cache.retrieveMetadata(imageReference)).thenReturn(Optional.of(imageMetadata));
     Mockito.when(
             cache.areAllLayersCached(imageMetadata.getManifestsAndConfigs().get(0).getManifest()))
@@ -423,7 +429,7 @@ public class PullBaseImageStepTest {
   }
 
   @Test
-  public void testGetCachedBaseImages_v22ManifestListCached_partialMatches()
+  public void testGetCachedBaseImages_manifestListCached_partialMatches()
       throws InvalidImageReferenceException, IOException, CacheCorruptedException,
           UnlistedPlatformInManifestListException, BadContainerConfigurationFormatException,
           LayerCountMismatchException, PlatformNotFoundInBaseImageException {
@@ -431,7 +437,7 @@ public class PullBaseImageStepTest {
     Mockito.when(buildContext.getBaseImageConfiguration())
         .thenReturn(ImageConfiguration.builder(imageReference).build());
 
-    V22ManifestListTemplate manifestList = Mockito.mock(V22ManifestListTemplate.class);
+    ManifestListTemplate manifestList = Mockito.mock(ManifestListTemplate.class);
     Mockito.when(manifestList.getDigestsForPlatform("arch1", "os1"))
         .thenReturn(Arrays.asList("sha256:digest1"));
     Mockito.when(manifestList.getDigestsForPlatform("arch2", "os2"))
@@ -442,7 +448,7 @@ public class PullBaseImageStepTest {
             manifestList,
             Arrays.asList(
                 new ManifestAndConfigTemplate(
-                    new V22ManifestTemplate(),
+                    Mockito.mock(BuildableManifestTemplate.class),
                     new ContainerConfigurationTemplate(),
                     "sha256:digest1")));
     Mockito.when(cache.retrieveMetadata(imageReference)).thenReturn(Optional.of(imageMetadata));
@@ -457,7 +463,7 @@ public class PullBaseImageStepTest {
   }
 
   @Test
-  public void testGetCachedBaseImages_v22ManifestListCached_onlyPlatforms()
+  public void testGetCachedBaseImages_manifestListCached_onlyPlatforms()
       throws InvalidImageReferenceException, IOException, CacheCorruptedException,
           UnlistedPlatformInManifestListException, PlatformNotFoundInBaseImageException,
           BadContainerConfigurationFormatException, LayerCountMismatchException {
@@ -465,7 +471,7 @@ public class PullBaseImageStepTest {
     Mockito.when(buildContext.getBaseImageConfiguration())
         .thenReturn(ImageConfiguration.builder(imageReference).build());
 
-    V22ManifestListTemplate manifestList = Mockito.mock(V22ManifestListTemplate.class);
+    ManifestListTemplate manifestList = Mockito.mock(ManifestListTemplate.class);
     Mockito.when(manifestList.getDigestsForPlatform("target-arch", "target-os"))
         .thenReturn(Arrays.asList("sha256:target-digest"));
 
@@ -474,10 +480,12 @@ public class PullBaseImageStepTest {
 
     ManifestAndConfigTemplate targetManifestAndConfig =
         new ManifestAndConfigTemplate(
-            new V22ManifestTemplate(), containerConfigJson, "sha256:target-digest");
+            Mockito.mock(BuildableManifestTemplate.class),
+            containerConfigJson,
+            "sha256:target-digest");
     ManifestAndConfigTemplate unrelatedManifestAndConfig =
         new ManifestAndConfigTemplate(
-            new V22ManifestTemplate(),
+            Mockito.mock(BuildableManifestTemplate.class),
             new ContainerConfigurationTemplate(),
             "sha256:unrelated-digest");
 

